js/interface.js

```diff
@@ -2111,6 +2111,34 @@ function getSubmissions() {
   return Fliplet.App.Submissions.get();
 }
 
+function setFirebaseStatus(credentialKey, origin) {
+  if (!credentialKey) {
+    return;
+  }
+  
+  var environment;
+
+  switch (origin) {
+    case 'appStore':
+      environment = 'store';
+      break;
+    case 'enterprise':
+      environment = 'env';
+      break;
+    case 'unsigned':
+      environment = 'uns';
+      break;
+    default:
+      break;
```

Comment:
default should be `Promise.reject('Invalid environment');`

Reply:
@squallstar  May I suggest using simple  `if` condition to handle environment error.
This function changes only a UI part. We do not need to wait until it ends.
That is why I wrote it without a `Promise`.
My suggestion is to set the `environment` variable to this value `Invalid environment`
And check for this value after a `switch`.
So it will look like that.
```
  switch (origin) {
    case 'appStore':
      environment = 'store';
      break;
    case 'enterprise':
      environment = 'env';
      break;
    case 'unsigned':
      environment = 'uns';
      break;
    default:
      environment = false;
      break;
  }

  if (!environment) {
    console.error('Invalid environment');
    return;
  }
```

Reply:
@YaroslavOvdii that looks good with me 👍

Reply:
@squallstar added an error handler.